grails-app/controllers/org/pih/warehouse/api/StocklistApiController.groovy

```diff
@@ -37,5 +37,66 @@ class StocklistApiController {
         render ([data:requisition] as JSON)
     }
 
+    def create = {
+        Requisition requisition = new Requisition()
 
+        JSONObject jsonObject = request.JSON
+        log.info "create " + jsonObject.toString(4)
+
+        bindStockList(requisition, jsonObject)
+
+        requisition.isTemplate = true
+        requisition = requisitionService.saveRequisition(requisition)
+        response.status = 201
```

Comment:
If previous line fail (won't save requisition), then this status should not be 201

Reply:
I think both of these scenarios are handled within the saveRequisition() and deleteRequisition() methods. These methods should throw a validation exception (or an authorization exception) which will get propagated to the exception handler (i.e. ErrorsController). But we should probably start adding some unit tests to be sure.

Reply:
I've been neglectful with respect to adding unit tests (and apologize for that) but felt it was too much overhead given how much work we had to do. Now that we might be getting back to a good place, I'd be happy to start working on unit tests for all the APIs.